src/libraries/Common/src/System/Reflection/AssemblyNameParser.cs

```diff
@@ -0,0 +1,524 @@
+        {
+#if NET8_0_OR_GREATER
+            ReadOnlySpan<char> attributeValueSpan = attributeValue;
+            Span<Range> parts = stackalloc Range[5];
+            parts = parts.Slice(0, attributeValueSpan.Split(parts, '.'));
+#else
+            string[] parts = attributeValue.Split('.');
+#endif
+            if (parts.Length is < 2 or > 4)
+            {
+                return false;
+            }
+
+            Span<ushort> versionNumbers = stackalloc ushort[4] { ushort.MaxValue, ushort.MaxValue, ushort.MaxValue, ushort.MaxValue };
+            for (int i = 0; i < parts.Length; i++)
+            {
+                if (!ushort.TryParse(
+#if NET8_0_OR_GREATER
+                    attributeValueSpan[parts[i]],
+#else
+                    parts[i],
+#endif
+                    NumberStyles.None, NumberFormatInfo.InvariantInfo, out versionNumbers[i]))
+                {
+                    return false;
+                }
+            }
+
+            if (versionNumbers[0] == ushort.MaxValue ||
+                versionNumbers[1] == ushort.MaxValue)
+            {
+                return false;
+            }
+
+            version =
+                versionNumbers[2] == ushort.MaxValue ? new Version(versionNumbers[0], versionNumbers[1]) :
+                versionNumbers[3] == ushort.MaxValue ? new Version(versionNumbers[0], versionNumbers[1], versionNumbers[2]) :
+                new Version(versionNumbers[0], versionNumbers[1], versionNumbers[2], versionNumbers[3]);
+
+            return true;
+        }
+
+        private static bool TryParseCulture(string attributeValue, out string? result)
+        {
+            if (attributeValue.Equals("Neutral", StringComparison.OrdinalIgnoreCase))
+            {
+                result = "";
+                return true;
+            }
+
+            result = attributeValue;
+            return true;
+        }
+
+        private static bool TryParsePKT(string attributeValue, bool isToken, ref byte[]? result)
+        {
+            if (attributeValue.Equals("null", StringComparison.OrdinalIgnoreCase) || attributeValue == string.Empty)
+            {
+                result = Array.Empty<byte>();
+                return true;
+            }
+
+            if (attributeValue.Length % 2 != 0 || (isToken && attributeValue.Length != 8 * 2))
+            {
+                return false;
+            }
+
+            byte[] pkt = new byte[attributeValue.Length / 2];
+            int srcIndex = 0;
+            for (int i = 0; i < pkt.Length; i++)
+            {
+                char hi = attributeValue[srcIndex++];
+                char lo = attributeValue[srcIndex++];
+
+                if (!TryParseHexNybble(hi, out byte parsedHi) || !TryParseHexNybble(lo, out byte parsedLo))
+                {
+                    return false;
+                }
+
+                pkt[i] = (byte)((parsedHi << 4) | parsedLo);
+            }
```

Comment:
Is this HexConverter.TryDecodeFromUtf16?

Reply:
It was, great catch!